src/context/internal.rs

```diff
@@ -208,67 +119,23 @@ pub trait ContextInternal<'a>: Sized {
     }
 }
 
-#[cfg(feature = "legacy-runtime")]
-extern "C" fn try_catch_glue<'a, 'b: 'a, C, T, F>(
-    rust_thunk: *mut c_void,
-    cx: *mut c_void,
-    returned: *mut c_void,
-    unwind_value: *mut *mut c_void,
-) -> TryCatchControl
-where
-    C: ContextInternal<'a>,
-    F: FnOnce(&mut C) -> NeonResult<T>,
-{
-    let f: F = *unsafe { Box::from_raw(rust_thunk as *mut F) };
-    let cx: &mut C = unsafe { &mut *cx.cast() };
-
-    // The mutable reference to the context is a fiction of the Neon library,
-    // since it doesn't actually contain any data in the Rust memory space,
-    // just a link to the JS VM. So we don't need to do any kind of poisoning
-    // of the context when a panic occurs. So we suppress the Rust compiler
-    // errors from using the mutable reference across an unwind boundary.
-    match catch_unwind(AssertUnwindSafe(|| f(cx))) {
-        // No Rust panic, no JS exception.
-        Ok(Ok(result)) => unsafe {
-            (returned as *mut T).write(result);
-            TryCatchControl::Returned
-        },
-        // No Rust panic, caught a JS exception.
-        Ok(Err(_)) => TryCatchControl::Threw,
-        // Rust panicked.
-        Err(err) => unsafe {
-            // A panic value has an undefined layout, so wrap it in an extra box.
-            let boxed = Box::new(err);
-            *unwind_value = Box::into_raw(boxed) as *mut c_void;
-            TryCatchControl::Panicked
-        },
-    }
-}
-
-#[cfg(feature = "legacy-runtime")]
-pub fn initialize_module(exports: Handle<JsObject>, init: fn(ModuleContext) -> NeonResult<()>) {
-    let env = Env::current();
-
-    ModuleContext::with(env, exports, |cx| {
-        let _ = init(cx);
-    });
-}
-
-#[cfg(feature = "napi-1")]
-pub fn initialize_module(
-    env: raw::Env,
-    exports: Handle<JsObject>,
+pub unsafe fn initialize_module(
+    env: *mut c_void,
+    exports: *mut c_void,
     init: fn(ModuleContext) -> NeonResult<()>,
 ) {
-    unsafe {
-        neon_runtime::setup(env);
-    }
+    let env = env.cast();
+
+    sys::setup(env);
 
     IS_RUNNING.with(|v| {
         *v.borrow_mut() = true;
     });
 
-    ModuleContext::with(Env(env), exports, |cx| {
+    let env = Env(env);
+    let exports = Handle::new_internal(JsObject::from_raw(env, exports.cast()));
```

Comment:
Correct! The `transmute` relied on the handle being a transparent wrapper around a pointer. Since `transmute` still enforces size, it wasn't that sketchy, but I still think this is a lot cleaner!